subprojects/logging/src/integTest/groovy/org/gradle/DeprecationHandlingIntegrationTest.groovy

```diff
@@ -129,6 +135,31 @@ class DeprecationHandlingIntegrationTest extends AbstractIntegrationSpec {
         'with stacktrace and --warning-mode=no'         | WarningMode.None    | 0                      | 0                      | true
         'without stacktrace and --warning-mode=summary' | WarningMode.Summary | 0                      | 4                      | false
         'with stacktrace and --warning-mode=summary'    | WarningMode.Summary | 0                      | 4                      | true
+        'without stacktrace and --warning-mode=fail'    | WarningMode.Fail    | 4                      | 0                      | false
+        'with stacktrace and --warning-mode=fail'       | WarningMode.Fail    | 4                      | 0                      | true
+    }
+
+    def 'build error and deprecation failure combined'() {
+        given:
+        buildFile << """
+            apply plugin: DeprecatedPlugin // line 2
```

Comment:
what's the purpose of `line 2`?

Reply:
Confirms that it matches the expected deprecation warning in the output. And again, pattern from that test file.